ABTesting

iliapir opened the pull request:
@anulman works with ABTesting branch of sheertex.com

iliapir committed:
Injected ABTests routes on GetStaticPaths for dynamic pages

iliapir committed:
Injected ABTests routes on GetStaticPaths for dynamic pages

iliapir committed:
Injected ABTests routes on GetStaticPaths for dynamic pages

iliapir committed:
Moved every component for ABTesting to next.js

iliapir committed:
Moved every component for ABTesting to next.js

iliapir committed:
Moved every component for ABTesting to next.js

Timer committed:
[Fast Refresh] Unlock Focus on Disabled Element (#12425)

matamatanot committed:
Use fs.promises in test files (#12432)

Timer committed:
v9.3.7-canary.7

Timer committed:
Fix linting

Timer committed:
v9.3.7-canary.8

Timer committed:
Do not prettify `lerna.json`

Timer committed:
v9.3.7-canary.9

anulman commented on packages/next/build/utils.ts:
```diff
@@ -756,7 +757,18 @@ export async function isPageStatic(
+        console.log(`Got static paths object -> ${JSON.stringify(staticPaths)}`)
+
+        // TODO: paths can have different params, not only 'handle' -> need to think what to do with other cases
+        const pathsWithAbTests = tryAddABTestingPayload(staticPaths.paths)
```
Can we call this function `permuteStaticPaths(staticPaths.paths)`

anulman replied:
How can we call this when `!pageIsDynamic || !hasStaticPaths || !hasStaticProps`?

anulman replied:
Really out there idea — if we exported `permuteStaticPaths(...)` from a `Page` file, could we run it via `mod.permuteStaticPaths`? The less code we add to mainline `next.js` the more likely our changes will be merged

anulman replied:
What if the function signature was `permuteStaticPaths: PermuteStaticPaths` (like the `GetStaticPaths` definition). where the type is:

```typescript
export type PermuteStaticPaths = (getStaticPaths?: GetStaticPaths): Array<{ params: ParsedUrlQuery }>
```

iliapir replied:
Love the idea of mod.permuteStaticPaths. We can avoid adding a lot of code to next.js using it.

anulman commented on packages/next/lib/abTestingInfra/Variant/index.tsx:
```diff
@@ -0,0 +1,21 @@
+function Variant(props: VariantProps): JSX.Element {
+  assertNameIsValid(props.name)
+
+  if (React.isValidElement(props.children)) {
+    return props.children
+  } else {
+    return <span>{props.children}</span>
+  }
```
Nice 👌

anulman commented on packages/next/lib/abTestingInfra/utils.ts:
```diff
@@ -0,0 +1,187 @@
+    ExperimentVariant
+  >
+
+  const result: Array<{ params: ParsedUrlQuery }> = []
+
+  paths.forEach(p => {
+    permutatedExperiments.forEach(pExp => {
+      const serializedAbTestingCtx = JSON.stringify(pExp) //next.js will do uriEncode for us
+
+      result.push({
+        params: {
+          handle: `${p.params.handle}${AB_TEST_PAYLOAD_PREFIX}${serializedAbTestingCtx}`,
+        },
+      })
+    })
+  })
+
+  return result
```
I see you using this pattern a lot — I challenge you to use JS' `reduce((acc, obj) => acc, defaultAcc)` instead

iliapir replied:
Will take it into account 👌

Timer committed:
Elements with `role=link` should open on [Enter] (#12460)

ijjk committed:
Add warning for non-standard NODE_ENV value (#12361)

ijjk committed:
Add additional file serving tests (#12479)

* Test `static/` file name encoding

* Fix `static/` file name encoding

* Add additional file-serving tests

* bump

Co-authored-by: Joe Haddad <joe.haddad@zeit.co>

Timer committed:
Upgrade Flexbugs Dependency (#12459)

Co-authored-by: JJ Kasper <jj@jjsweb.site>

devknoll committed:
Don't generate fake cache keys for Preview Mode (#12487)

jokester committed:
next/client/link: improve internal type annotation (#12468)

Timer committed:
[Fast Refresh] New Overlay for Prerender Error (#12485)

Timer committed:
v9.3.7-canary.10

piglovesyou committed:
[examples/with-typescript-graphql] Add "createResolverContext" hook, etc. (#11681)

* Update deps

* Run "node-check-update -u"
* "graphql" keeps ^14 since other libs expect it
* Install apollo-link and apollo-cache because @apollo/react-common
wants it as peer-deps

* Add graphql-let/schema/loader

* This enables HMR from modifying *.graphqls.

* Add resolver context

* Remove JSDoc, respect TypeScript defs

* ncu -u

* Keep using graphql@^14.6.0 for the other deps

* fix: Migrate to graphql-let@0.10.0

* Removed duplicated deps

* Updated check

Co-authored-by: Luis Alvarez <luis@zeit.co>

plug-n-play committed:
Fixed typo (#12497)

ijjk committed:
Add note about HoC possibly adding GIP to _app (#12491)

Timer committed:
Disable cache warning on supported platforms (#12495)

Timer committed:
[Fast Refresh] Better Tabbing Order (#12503)

Timer committed:
[Fast Refresh] Correctly Trap Focus in Overlay (#12502)

* [Fast Refresh] Correctly Trap Focus in Overlay

* Adjust escape behavior to be more correct

Co-authored-by: JJ Kasper <jj@jjsweb.site>

Timer committed:
Disable Comment Extraction (#12496)

Timer committed:
[Fast Refresh] Fix overlay display on Safari (#12523)

Timer committed:
v9.3.7-canary.11

ijjk committed:
Update tests to poll for output instead of wait for it (#12534)

devknoll committed:
Don't add `.amp` to pathname (#12528)

austinhumes-valtech committed:
Update dynamic-import.md (#12541)

Timer committed:
[Fast Refresh] New Compile Error Overlay (#12543)

Timer committed:
v9.3.7-canary.12

arcanis committed:
Adds missing deps (#12519)

timneutkens committed:
Add documentation for TypeScript baseUrl and paths option (#12379)

* Add documentation for TypeScript baseUrl and paths option

* Update module-path-aliases.md

* Update manifest.json

mlorthiois committed:
Update blog-starter to use Tailwind CSS v1.4's new built-in purge option (#12452)

huozhi committed:
fix: hello world missing default exports (#12551)

lachlanjc committed:
Mention nesting in arrays alongside Fragments in next/head docs (#12420)

Co-authored-by: Tim Neutkens <tim@timneutkens.nl>
Co-authored-by: Joe Haddad <joe.haddad@zeit.co>

iliapir committed:
Added permuteStaticPaths

iliapir committed:
Added permuteStaticPaths

sebastianbenz committed:
Update AMP dependencies (#12536)

Timer committed:
[Fast Refresh] Toggling of Collapsed Frames (#12560)

Timer committed:
[Fast Refresh] Separate LogBox Tests (#12564)

AlexanderVishnevsky committed:
Resolve bug with Prismic Locales (#12132)

* Resolve bug with Prismic Locales

add configuration for not default locale

* Resolve bug with Prismic Locales

Add configuration for Prismic locale.
More information about bug:

* Fixes and removed unrequired changes

Co-authored-by: Luis Alvarez <luis@zeit.co>

Timer committed:
[Fast Refresh] Add more test cases (#12568)

rafaelalmeidatk committed:
Export NextApiRequest type as interface (#12186)

mxschmitt committed:
example: added Ant Design Pro Layout less (#12408)

ijjk committed:
Update to run tests in headless mode on Azure (#12574)

ijjk committed:
Fix `moduleId` not resolving correctly on Windows (#12584)

Timer committed:
v9.3.7-canary.13

janicklas-ralph committed:
Experimental: New ESLint Plugin (#11986)

Timer committed:
v9.3.7-canary.14

NavizDev committed:
Add cross-env package (#12581)

* add notes

* add cross-env package, for stability in different SO

* fix: move cross-env package from Dependencies at devDependicies

PabloSzx committed:
Add flag to disable API warning (#11380)

* Add flag to disable API warning

This flag is useful when you are using an external API resolver like express when defining an API route, since the native functionality doesn't realize that the API actually sent a response.

A very simple use case example https://github.com/PabloSzx/next-external-api-resolver-example

fixes #10589

* Update api-middlewares.md

Co-authored-by: Tim Neutkens <tim@timneutkens.nl>

ijjk committed:
Update to latest version of stacktrace-parser (#12597)

anulman commented on packages/next/build/index.ts:
```diff
@@ -806,13 +806,36 @@ export default async function build(dir: string, conf = null): Promise<void> {
         if (!isDynamic) {
-          await moveExportedPage(page, file, true, 'json')
+          if (additionalSsgPaths.has(page)) {
+            // TODO: maybe we should move it to a separate function? see duplicated code below
```
+1 agreed! What function signature would you propose?

iliapir2 replied:
`async addExtraRoutesForPage(page:string, extraRoutes:string[]): Promise<void>`

~~It will be function with side effects, as it will change finalPrerenderRoutes internally~~

Check comment below

anulman reviewed (commented):
Looks good! I agree re refactoring, but this is a nice tight change

TyMick committed:
Fix DatePicker Day.js bug in Ant Design example (#12580)

* Fix DatePicker Day.js bug in Ant Design example

* Lint fix

Co-authored-by: Luis Alvarez <luis@vercel.com>

lwz7512 committed:
[Examples]: With react bootstrap (#12421)

* add new example with-react-bootstrap

* use vanilla react-bootstrap components to implement index page

* clear unnecessary comments

* remove private property in package.json

* Updates

Co-authored-by: Luis Alvarez <luis@vercel.com>

Timer committed:
Fix error toast pluralization (#12622)

Timer committed:
Clarify dismissal message (#12624)

Timer committed:
[Fast Refresh] Adjust Overlay Styling (#12623)

Timer committed:
v9.3.7-canary.15

Commit:
Context types for getStaticProps and getServerSideProps (#12606)

Timer committed:
[Fast Refresh] Upgrade `react-refresh` for SC (#12627)

Timer committed:
v9.3.7-canary.16

lfades committed:
[Examples] Update data-fetch example (#12617)

Timer committed:
[Fast Refresh] Overlay hash must consider stack (#12639)

Skn0tt committed:
Allow `@babel/preset-typescript` to be configured (#11840)

* Allow `@babel/preset-typescript` to be configured

Some usages, e.g. [getting parameter decorators to work properly](https://github.com/WarnerHooh/babel-plugin-parameter-decorator/pull/19), require configuring `@babel/preset-typescript`.

This commit adds that functionality to `next/babel`.

* Add unit test for babel/preset allowing to pass options to @babel/preset-typescript

* Add integration test for onlyRemoveTypeImports

* Update babel dependencies

* Update to compatible typescript version and fix types

* Fix linting and run pre-nccing

* Update size-limit test

* Add additional tests

* Re-Apply delta to be calculated using -262

Co-authored-by: JJ Kasper <jj@jjsweb.site>

chibicode committed:
More renaming ZEIT → Vercel (#12649)

ijjk committed:
Disable experimental pageEnv stub by default (#12644)

ijjk committed:
Cont. Update performance relayer and remove unstable prefix (#12665)

* updates relayer, uses web-vital lib, removes fid polyfill and removes unstable prefix from relayer

* simplify logic, fix null check

* update docs

* Update tests

* Rename relayWebVitals to reportWebVitals

Co-authored-by: housseindjirdeh <houssein@google.com>

Timer committed:
[Fast Refresh] Reorder Loaders (#12671)

Timer committed:
[Fast Refresh] Resize Arrows and Preserve Newlines (#12672)

Timer committed:
[Fast Refresh] Do not draw eye to collapsed frames (#12673)

Timer committed:
v9.3.7-canary.17

eps1lon committed:
fix: use latest experimental method names (#12669)

Timer committed:
Fix check utility in tests (#12678)

timneutkens committed:
v9.3.7-canary.18

maxigimenez committed:
chore(create-next-app): add missing return types on helpers (#12616)

Co-authored-by: Tim Neutkens <tim@timneutkens.nl>

Timer committed:
[Fast Refresh] Add additional test case (#12674)

timneutkens committed:
Update documentation that refers to isomorphic fetch libs (#12686)

maxigimenez committed:
chore(create-next-app): add missing types (#12691)

maxigimenez committed:
chore(next): link/router/document missing types (#12692)

Timer committed:
Enable Fast Refresh by Default (#12640)

Timer committed:
Upgrade deps (#12700)

Timer committed:
v9.3.7-canary.19

lfades committed:
[Examples] Update firebase authentication (#12687)

timneutkens committed:
Update code sample in performance measurement docs (#12690)

timneutkens committed:
Enable sassOptions support in next.config.js (#12685)

Timer committed:
v9.3.7-canary.20

sarupbanskota committed:
Link Made by Vercel to vercel.com (#12702)

lfades committed:
[Docs] Environment Variables (#12696)

belgattitude committed:
Use @/ to show module alias instead of @ prefixed (#12600)

* docs: controversial '@' prefix aliases should not be used

* Update docs/advanced-features/module-path-aliases.md

Co-authored-by: Joe Haddad <timer150@gmail.com>

* Update docs/advanced-features/module-path-aliases.md

Co-authored-by: Joe Haddad <timer150@gmail.com>

Co-authored-by: Tim Neutkens <tim@timneutkens.nl>
Co-authored-by: Joe Haddad <timer150@gmail.com>

Commit:
fix logout in with-iron-session example (#12704)

timneutkens committed:
Add test for API route with fetch polyfill (#12680)

Timer committed:
v9.3.7-canary.21

Timer committed:
v9.4.0

Commit:
Minor update to "Measuring performance" docs (#12741)

sergiodxa committed:
Add missing parenthesis in Sending results to analytics example (#12733)

ijjk committed:
Add message when automatic dotenv loading is disabled (#12744)

Timer committed:
[Fast Refresh] Cleaner Console Output (#12745)

0xflotus committed:
(docs): fixed typo error (#12746)

Timer committed:
v9.4.1-canary.0

jamiebuilds committed:
Use an if statement instead of expression statement (#12747)

rafaelalmeidatk committed:
Fix typo in Environment Variables doc page (#12770)

iliapir committed:
Added support of specific page names that equals actual page name

iliapir committed:
Added support of specific page names that equals actual page name

iliapir committed:
Added support of specific page names that equals actual page name

iliapir2 commented on packages/next/build/index.ts:
```diff
@@ -809,7 +815,12 @@ export default async function build(dir: string, conf = null): Promise<void> {
             const extraRoutes = additionalSsgPaths.get(page) || []
-            for (const route of extraRoutes) {
+
+            for (let route of extraRoutes) {
+              if (route === '/') {
+                route = 'index'
+              }
+
```
@anulman I added extra code, so now it's not that easy to refactor this duplicated code

T0MASD committed:
sort posts by date in descending order (#12303)

* sort posts by date in descending order

Hi, I'm extending getAllPosts function to sort posts array by date in descending order. 
Any feedback is appreciated

* sort posts by date timestamp in descending order

* Lint fix

Co-authored-by: Luis Alvarez <luis@vercel.com>

timothyis committed:
Add contentful as a related example link to each CMS example (#12796)

timneutkens committed:
Make fetch polyfill work with ky-universal (#12804)

Timer committed:
v9.4.1-canary.1

shunkakinoki committed:
docs: redirect error to the correct url (#12813)

jpedroschmitz committed:
docs(ssr-caching): remove link to react-esi example (#12789)

Timer committed:
Bump Babel cache key (#12822)

Timer committed:
Run all logbox tests (#12825)

timneutkens committed:
Fix edge case in fetch polyfill (#12832)

* Add tests for getStaticPaths

* Fix edge case in fetch polyfill

Fixes #12794

* Correct typo in tests

timneutkens committed:
v9.4.1-canary.2

acifani committed:
chore(deps): upgrade url-polyfill (#12829)

Timer committed:
Better Babel syntax errors (#12821)

Timer committed:
Fix up CMS examples (#12840)

Timer committed:
Simplify CSS Errors (#12852)

Timer committed:
v9.4.1-canary.3

dylanjha committed:
Add example with-hls-js (#12723)

* npm init next-app

* add example video-player component with HLS.js

* add README for example with HLS.js

* update name in package.json

* Update examples/with-hls-js/.gitignore

Co-authored-by: Luis Alvarez D. <luis@vercel.com>

* Update license to MIT

Co-authored-by: Luis Alvarez D. <luis@vercel.com>

* Update next version to latest

Co-authored-by: Luis Alvarez D. <luis@vercel.com>

* remove unused pages/api

Co-authored-by: Luis Alvarez D. <luis@vercel.com>

Timer committed:
Eliminate 404s from internal requests (#12871)

giuseppeg committed:
Update styled-jsx to 3.3.0 (#12870)

Introduces a new plugin option to configure the module that the transpiled files should import. 
Should unblock https://github.com/zeit/next.js/issues/10149 and support for yarn v2 once #10149 is complete.

https://github.com/zeit/styled-jsx/releases/tag/v3.3.0

Timer committed:
Remove `isInitialRender` usage for overlay (#12872)

iliapir committed:
Added path to params on non Dynamic pages to later on extract abTestingPayload in getStaticProps handler

iliapir committed:
Added path to params on non Dynamic pages to later on extract abTestingPayload in getStaticProps handler

iliapir committed:
Added path to params on non Dynamic pages to later on extract abTestingPayload in getStaticProps handler

briangershon committed:
Warn users about migrating to 9.4 from dotenv (#12859)

Timer committed:
Explicit header for SSR error (#12880)

Timer committed:
Scroll entire content instead of only header (#12881)

Timer committed:
v9.4.1-canary.4

raphaelchaula committed:
Update package.json (#12865)

Without babel-core, "lingui extract" gives TypeError: Api.call... function is undefined

EvanBacon committed:
add Expo example (#12805)

* Created Expo example project

* Update examples/with-expo/README.md

* Updated usage guide

* Update README.md

* Lint fix

Co-authored-by: Luis Alvarez <luis@vercel.com>

Timer committed:
`sync` and `built` are identical commands (#12908)

nicholaschiang committed:
Update module-path-aliases.md (#12900)

Timer committed:
[Fast Refresh] Improve responsiveness on Windows (#12909)

Timer committed:
v9.4.1-canary.5

brunocrosier committed:
Stop tracking changes in .next folder (#12924)

The .next folder doesn't need to be tracked

HaNdTriX committed:
[Examples] Fix active-class-name (#12931)

ijjk committed:
Correct multi-match param behavior (#12930)

Timer committed:
Strip out dead code for `ignoreDevErrors` (#12941)

Timer committed:
Remove `react-error-overlay` (#12942)

timneutkens committed:
Enable .env support by default (#12911)

* Enable .env support by default

Given we've had tons of reports from various people that expected .env support to work even though they had dotenv installed already I think it's fine to enable it as a default:

Fixes #12728

* Remove old test

* Fix duplicate env loading

* Update docs

Co-authored-by: JJ Kasper <jj@jjsweb.site>

ijjk committed:
v9.4.1-canary.6

ijjk committed:
Add retrying to cross browser tests (#12946)

willianjusten committed:
Remove `isomorphic-unfetch` from examples (#12948)

Since 9.4 release, fetch is pollyfilled by default from #12353,
so the import is not needed anymore.

Timer committed:
Elimination has to run before other plugins (#12949)

Timer committed:
Support eliminating memoized functions (#12953)

ijjk committed:
Fix build stalling with styled-jsx plugins (#12952)

Timer committed:
v9.4.1-canary.7

Timer committed:
Test TypeScript Enum (#12954)

ijjk committed:
Add experimental Storybook preset (#12934)

lfades committed:
Use a top level heading (#12960)

Janpot committed:
Use the jest-circus test runner (#12974)

timneutkens committed:
Bypass OPTIONS handling for API routes (#12978)

Timer committed:
v9.4.1-canary.8

none23 committed:
Remove unused babel-preset-react dep from babel-macros example (#12990)

Timer committed:
Update Babel Cache Key (#12989)

Timer committed:
v9.4.1-canary.9

Timer committed:
v9.4.1

Janpot committed:
Only retry tests on CI (#12998)

timneutkens committed:
[Experimental] Add support for production browser source maps (#13018)

felixmosh committed:
fix modal overlay in RTL layouts (#13000)

teo-garcia committed:
feat: Remove redundant imports in XState example (#13020)

Timer committed:
Convert test to be inline (#13022)

developit committed:
Avoid over-transpiling core libs (#11803)

* Avoid over-transpiling core libs

* Update preset-env to 7.9

Co-authored-by: Joe Haddad <joe.haddad@zeit.co>

Timer committed:
v9.4.2-canary.0

Janpot committed:
[Experimental] Implement optional catch all routes (#12887)

Timer committed:
v9.4.2-canary.1

kolyasapphire committed:
[Examples] Chakra UI - Move CSSReset inside ColorModeProvider (#13044)

Timer committed:
Stabilize test for Windows (#13050)

Janpot committed:
Improve jsconfig.json error message (#13053)

Timer committed:
An empty `jsconfig.json` should not fail the build (#13051)

rafaelalmeidatk committed:
Add PR labeler action (#13054)

Timer committed:
Revert "Add PR labeler action" (#13055)

stigkj committed:
x-powered-by is added to response, not request (#13037)

teo-garcia committed:
chore: Remove redundant imports in several examples (#13030)

ijjk committed:
Correct multi-match behavior for queries and header values (#13017)

lboecker committed:
Fix with-emotion examples (#13005)

Commit:
Remove redundant React imports from with-redux-thunk example (#12986)

Commit:
[Examples] Added public folder to with-docker example (#12879)

b-bot committed:
docs: add prismic locale note (#12862)

ubmit committed:
chore: use function component on with-typescript example (#12695)

Commit:
fix crash when NODE_OPTIONS includes --inspect-port (#12555)

Janpot committed:
Add eslint-plugin-jest (#13003)

Ehesp committed:
Add monaco-editor example (#11232)

maxigimenez committed:
chore: hot-reloader, htmlescape, next-dev-server missing types (#12730)

Timer committed:
Fix lint

todortotev committed:
Removed redundant code from the with-portals example (#13059)

Timer committed:
Upgrade to Prettier 2 (#13061)

Timer committed:
Update Examples for Fast Refresh (#13068)

Timer committed:
v9.4.2-canary.2

timneutkens committed:
Add sourcemaps to Next.js core compilation (#13049)

* Generate sourcemaps for core files that pass through Babel

* Run files through Babel instead of tsc

* Get rid of wildcard helper

* Get rid of wildcard helper

* Remove unused file

* Update wildcard imports

* Add exclude

* Get rid of object-assign helper

* Use Object.assign as it gives better output

Co-authored-by: Joe Haddad <joe.haddad@zeit.co>

prateekbh committed:
adding no html-link lint rule to eslint-plugin (#12969)

* addinng no html-link lint rule

* fixing lint tests

* adding the utils file

* fixing lock file

* prettier fix

maxigimenez committed:
chore(next): check-custom-routes - recursive-copy missing types (#13065)

timneutkens committed:
Remove taskr typescript plugin as it's no longer used (#13081)

lfades committed:
Added path (#13075)

todortotev committed:
Remove redundant code from with-portals-ssr example (#13060)

* removed redundant imports and replaced the fragment with the shorthand syntax

* Brought back the component import

Co-authored-by: Joe Haddad <joe.haddad@zeit.co>

timneutkens committed:
Update prefetching explanation to correctly reflect data prefetching (#13088)

slawekkolodziej committed:
Fix building server-side generated AMP pages (#13046)

Timer committed:
v9.4.2-canary.3

leosuncin committed:
Improve Redux Toolkit example (#12858)

ijjk committed:
Add error when exporting pages with fallback: true (#13063)

zlwaterfield committed:
Update the Automatic Static Optimization docs to explicitly state that Next.js will trigger an update. (#13096)

twltwl committed:
fix(examples): align with-absolute-imports with next 9.4 (#13090)

cjimmy committed:
[Example] Unsubscribe Router.on.events (#13102)

Timer committed:
Add Kodiak Configuration (#13103)

Janpot committed:
Fix catch-all route + index.js in dev when accessed with trailing slash (#10502)

Failing test case for https://github.com/zeit/next.js/issues/10488#issuecomment-584500081

This used to give a 500 in dev environment

Timer committed:
Fix Lint

Timer committed:
Stabilize `config` tests (#13116)

gregrickaby committed:
Update DatoCMS Example (#13008)

👋🏻 This PR updates the Datocms example:

- [x] Next.js 9.4 absolute import and aliases support
- [x] Next.js 9.4 native `.env` support
- [x] Tailwind 1.4 native PurgeCSS support
- [x] Chore: bump other dependencies, like React, Remark, and DateFNS 

I've tested this PR here: 

- 💻 https://github.com/gregrickaby/datocms-next-js-blog-demo-2392
- 👀 https://datocms-next-js-blog-demo-2392.now.sh/

Timer committed:
Enable `update_branch_immediately`

Timer committed:
Revert "Enable `update_branch_immediately`"

This reverts commit b0a020e8ef229cc324cbfd95540180044ed435bd.

lfades committed:
[Examples] Update with-sentry-simple (#13074)

Moved from `getInitialProps` to the new data fetching methods. Updated the readme and simplified the implementation for pages.

timneutkens committed:
Add support for debugging Next.js core (#13101)

Note this is only for the Next.js core, will look at a configuration for the debugger for Next.js apps soon.

timneutkens committed:
Read process.env after loading .env files (#13087)

Fixes #13067

davecardwell committed:
[Experimental] Only consider files within plugins (#12943)

I have a plugin with a src folder structure following jest’s recommended practice of including a `__tests__` folder:

```
$ find src/
src/
src/document-html-props-server.ts
src/__tests__
src/__tests__/document-html-props-server.unit.ts
```

collect-plugins.ts [reads the src directory](https://github.com/zeit/next.js/blob/960c18da53af89115c7e96dc8188646a51c53c15/packages/next/build/plugins/collect-plugins.ts#L65) and finds:
```js
[ '__tests__', 'document-html-props-server.ts' ]
```

…which it then [converts to](https://github.com/zeit/next.js/blob/960c18da53af89115c7e96dc8188646a51c53c15/packages/next/build/plugins/collect-plugins.ts#L76-L80):
```js
[ '', 'document-html-props-server' ]
```

…and finally [outputs](https://github.com/zeit/next.js/blob/960c18da53af89115c7e96dc8188646a51c53c15/packages/next/build/plugins/collect-plugins.ts#L90-L96):
```
Next.js Plugin: {my plugin} listed invalid middleware
```

This pull request makes collect-plugins.ts only consider files, so my __tests__ directory and its contents are ignored.

taye committed:
example(contentful): set up content types with required fields (#12591)

The example app expects all fields of Post and Author entries to be present and fails to build if any are missing. This PR adds a "required" validation to fields of both content types to avoid bulid errors.

CC @stefanjudis

Prioe committed:
[Examples] Fix with-yarn-workspaces Vercel deploy button (#12867)

Fixes: https://github.com/zeit/next.js/issues/10734
Related: https://github.com/zeit/now/discussions/4320

This fixes the **Deploy your own** button for the with-yarn-workspaces example. Currently the deployment fails with the following error.

```
Module not found: Can't resolve 'bar' in '/zeit/3ad1f9cd/pages'
```

I also added a dynamic api route to the example, because I had problems configuring vercel to redirect to dynamic routes.

This will be the new Vercel-Import-From link: [Deploy](https://vercel.com/import/project?template=https://github.com/Prioe/next.js/tree/fix-with-yarn-workspaces-deploy/examples/with-yarn-workspaces)

Timer committed:
Use `eval-source-map` for Server Side Errors (#13123)

This switches to faster source maps in development for the server-side compilation on macOS.

We still need to figure out a story for Windows.

teo-garcia committed:
chore: Remove some redundant imports (#13066)

This issue is related to [12694](https://github.com/zeit/next.js/issues/12964). I covered the following examples


- with-zeit-fetch
- with-yarn-workspaces
- with-why-did-your-render
- with-video-js
- with-universal-configuration-runtime
- with-typestyle
- with-three-js

If you have a suggestion or change I'd appreciate it

christianalfoni committed:
With stitches (#12677)

split into two examples

janicklas-ralph committed:
GranularChunks conformance check (#11710)

Adding a conformance plugin the make sure users don't undo the benefits of the granularChunks config.

The plugin makes sure that minSize, maxInitialRequests values aren't overridden. Also ensures the cacheGroups - vendors, framework, libs, common, shared are maintained.

The warning and error messages do not break the build with this change. They only display a message.

cc - @prateekbh, @atcastle

Timer committed:
Enable jest/no-try-expect (#13124)

kirill-konshin committed:
Upgrade Next Redux Wrapper to version 6 in examples (#11802)

Fixes https://github.com/zeit/next.js/issues/12920

pkuczynski committed:
Fix the link for Custom Server documentation (#13137)

Timer committed:
Correctly eliminate destructuring assignment (#13144)

This eliminates code referenced via destructuring assignment, as reported by @styfle.

Timer committed:
v9.4.2-canary.4

ijjk committed:
Update Create Next App to handle rename of org (#13138)

* Update Create Next App to handle rename

* Rethrow if we dont handle the error

Timer committed:
Eliminate array destructuring assignment (#13151)

While we're at it, we should fix this too.

x-ref: #13144

Timer committed:
Fix Windows LogBox Test (#13147)

todortotev committed:
Feat/remove redundant imports from examples (#13062)

The issue is related to [12964](https://github.com/zeit/next.js/issues/12964)

Let me know if there are any changes you want me to make.

Affected examples:

**with-glamor
with-graphql-hooks
with-graphql-react
with-grommet
with-http2
with-jest
with-cookie-auth-fauna
with-context-api
with-cerebral
with-aphrodite
with-apollo-and-redux
basic-css
with-carbon-components
amp-first**

I would love to help more, so let me know if there is anything specific I can contribute to.

Timer committed:
Record presence of `reportWebVitals` (#13155)

Closes #12897

Timer committed:
v9.4.2-canary.5

Timer committed:
v9.4.2

devknoll committed:
Remove dead events (#13156)

devknoll committed:
Remove unnecessary loadGetInitialProps (#13159)

Timer committed:
Remove old ErrorComponent global

luciovilla committed:
Load .env files before running Telemetry (#13173)

Fixes #13171

ijjk committed:
Migrate PR stats action into Next.js repo (#13177)

* Migrate PR stats into Next.js repo

* Update running prettier in local mode

iliapir committed:
Merge branch 'ABTesting' of https://github.com/Sheertex/next.js

# Conflicts:
#	packages/next/types/index.d.ts

iliapir committed:
Merge branch 'ABTesting' of https://github.com/Sheertex/next.js

# Conflicts:
#	packages/next/types/index.d.ts

iliapir committed:
Merge branch 'ABTesting' of https://github.com/Sheertex/next.js into ABTesting

# Conflicts:
#	packages/next/build/index.ts
#	packages/next/types/index.d.ts

maxigimenez committed:
chore(next/build/babel): missing types (#13189)

iliapir committed:
Merge branch 'ABTesting' of https://github.com/Sheertex/next.js into ABTesting

# Conflicts:
#	packages/next/build/index.ts
#	packages/next/build/utils.ts

iliapir committed:
Merge fix

iliapir2 commented:
Moved everything to ABTesting2
